### DISCUSSION TITLE
Default poster broken and how to add multiple source qualities?

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vidstack/player#983] [Q&A] [answered]
![image](https://github.com/vidstack/player/assets/33315834/8e8a37db-5287-464a-8e34-c04f95ad89ac)

I want to add poster to the default layout, but it seems not working as expected.

```ts
    return (): VNode =>
      h("media-player", attrs, [
        h("media-provider", [
          attrs["poster"]
            ? h("media-poster", { alt: attrs["alt"] || attrs["title"] })
            : null,
          props.sources.map((source) =>
            isPlainObject(source)
              ? h("source", source)
              : h("source", { src: source })
          ),
          props.tracks.map(
            ({ src, label, srclang, kind, default: isDefault }) =>
              h("track", { src, label, srclang, kind, default: isDefault })
          ),
          h("media-gesture", { event: "pointerup", action: "toggle:paused" }),
          h("media-gesture", {
            event: "dblclick",
            action: "toggle:fullscreen",
          }),
        ]),
        h("media-audio-layout"),
        h("media-video-layout"),
      ]);
  },
```

**Comment by mihar-22** [comment, marked as answer]
Hey @Mister-Hope!

1. If you'd like the default styles applied to the poster, include the `vds-poster` class on it:

```html
<media-poster class="vds-poster">
```
2. Gestures are already included by default in the video layout.

3. Multiple sources require a `type` hint to determine which to load:

```html
<source src="https://media-files.vidstack.io/720p.ogv" type="video/ogg" />
<source src="https://media-files.vidstack.io/720p.avi" type="video/avi" />
<source src="https://media-files.vidstack.io/720p.mp4" type="video/mp4" />
```

**Reply by Mister-Hope** [reply]
THanks, I also have other problems, plyr layout is available, but I seems not find a automatically generated quality menu when providing multiple sources:

```ts
const sources = [
  {
    src: "https://cdn.plyr.io/static/demo/View_From_A_Blue_Moon_Trailer-576p.mp4",
    type: "video/mp4",
    size: 576,
  },
  {
    src: "https://cdn.plyr.io/static/demo/View_From_A_Blue_Moon_Trailer-720p.mp4",
    type: "video/mp4",
    size: 720,
  },
  {
    src: "https://cdn.plyr.io/static/demo/View_From_A_Blue_Moon_Trailer-1080p.mp4",
    type: "video/mp4",
    size: 1080,
  },
  {
    src: "https://cdn.plyr.io/static/demo/View_From_A_Blue_Moon_Trailer-1440p.mp4",
    type: "video/mp4",
    size: 1440,
  },
];
```

Is there extra job I shall do?

**Reply by mihar-22** [reply]
I can see the value in adding support for this as an initial load option (similar to the `<picture>` element for images),  but I'm not sure we'll support dynamically switching qualities using static media containers like that (e.g., showing up in quality selector). That might've been a fallback before ~2017, but tech such as HLS and native `MediaStream` have evolved and is much more reliable and accessible. HLS is a much more effective/efficient way of performing adaptive streaming then multiple static containers (e.g., MP4). For example, with multiple MP4 videos you can't switch quality mid-playback without loading an entirely new MP4 and the media itself will completely reset. We would then have to write logic to reset playback correctly. 

If you'd like to create a HLS playlist yourself, you can use something like ChatGPT to help write a FFMPEG commands to take your highest quality video (e.g., 1080p) and build it, then dump it in a static storage just like your existing MP4 files. Another option is to rely on a service out there like Mux, Cloudflare Stream, Bunny Stream, or whatever else to do it for you which costs money but is easier/faster.

Honestly, the cost and effort to create and host a HLS playlist today is equal to that of multiple MP4 resolutions.

**Reply by mihar-22** [reply]
I'm definitely happy to try looking into adding the initial load option this weekend for multiple sources with a `size` property.
